File: attribouter/src/main/java/me/jfenn/attribouter/wedges/link/EmailLinkWedge.java
package me.jfenn.attribouter.wedges.link;

import android.content.res.XmlResourceParser;
import androidx.annotation.NonNull;

public class EmailLinkWedge extends LinkWedge {

    public EmailLinkWedge(XmlResourceParser parser) {
        this(parser.getAttributeValue(null, "email"), 0);

        String priorityString = parser.getAttributeValue(null, "priority");
        if (priorityString != null)
            priority = Integer.parseInt(priorityString);
    }

    public EmailLinkWedge(@NonNull String address, int priority) {
        super("email", "@string/title_attribouter_email", "mailto:" + address, "@drawable/ic_attribouter_email", false, priority);
    }

}

File: attribouter/src/main/java/me/jfenn/attribouter/wedges/link/EmailLinkWedge.kt
package me.jfenn.attribouter.wedges.link

class EmailLinkWedge(
        address: String, priority: Int
) : LinkWedge(
        id = "email",
        name = "@string/title_attribouter_email",
        url = "mailto:$address",
        icon = "@drawable/ic_attribouter_email",
        priority = priority
)
